### PR TITLE
docs: initial commit

### PR DESCRIPTION
**Pull request by khamui**

### CODE REVIEW

**Comment by khamui** [comment]
@phillimorland help for my bad english very welcome :) feel free to add or change the content. I just wrote whatever i remember we talked about while developing.

**Review by fuzzylogic2000 — APPROVED** [review]
Cool!

File: docs/coding_guidelines.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+**Single Responsability**
+Sometimews it happens, that developers write complex functions, components and classes that do a lot of different things.
+These code bases are hard to maintain. Bugs are harder to find and to be fixed.
+That is why we stick to this principle and keep the responsability of functions, components and classes as less as possible.
+
+**DRY - Don't Repeat Yourself**
+If it happens, that we find similar or same functionality in more than one spot, we should keep consolidating them.
+That helps in case of modification in future. We then only have to adjust the code in one spot.
+
```
Comment (phillimorland) [inline docs/coding_guidelines.md:73]: Note: should have a policy of adding `FIXME` anywhere we find repeated code if we don't have time for a refactor in that moment? 
Also just as a comment I saw this as problem with styles alot, i'm not sure how to fix that other then make everything in tiny components, is there another way? can we have a utilities style sheet that can be imported into other style sheets or something? maybe we can consider this for next sprint

File: docs/coding_guidelines.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+“Dumb components are also called ‘presentational’ components because their only responsibility is to present something to the DOM. Once that is done, the component is done with it. No keeping tabs on it, no checking in once in a while to see how things are going. Nope. Put the info on the page and move on.”
+
+Container components:
+
+“Smart components (or container components) on the other hand have a different responsibility. Because they have the burden of being smart, they are the ones that keep track of state and care about how the app works. Using the container design pattern, the container components are separated from the presentational components and each handles their own side of things. The container components do the heavy lifting and pass the data down to the presentational components as props.”
+
+Source of excerpts: https://medium.com/@thejasonfile/dumb-components-and-smart-components-e7b33a698d43
+
+**Single Responsability**
+Sometimews it happens, that developers write complex functions, components and classes that do a lot of different things.
```
Comment (phillimorland) [inline docs/coding_guidelines.md:66]: *sometimes

**Review by phillimorland — COMMENTED** [review]
Really cool! :)

**Pull request merged**